Component analyzer upgrade

vicdoval opened the pull request:
In the component analyzer :
Numpy routines for Faces and Vertex Normals, Centers, Perimeter and Tangent Edge
This are the results of some test
```

Icosphere 8 subdivisions
    center median
        Old     0.682  
        New     0.224  
    center bbox
        Old     0.665  
        New     0.369 
    center weighted
        Old     0.668  
        New     0.375 
        
    normals
        Old     0.621  
        New     0.316  
    normals absolute
        Old     0.697  
        New     0.353  

    perimeter
        Old     0.615  
        New     0.182  

    tangent edge
        Old 0.668
        New 0.321

10000 faces voronoi sphere (variable polygon sides)
    center median
        Old     0.037  
        New     0.018  
    center bbox
        Old     0.035  
        New     0.026  
    center weighted
        Old    0.033  
        New    0.022  

    normals
        Old     0.032  
        New     0.020  
    normals absolute
        Old     0.034  
        New     0.028  

    perimeter
        Old     0.031  
        New     0.017  

    tangent edge
        Old     0.034
        New     0.018

    vertex normals
        Old     0.036
        New     0.025
```
## Preflight checklist

Put an x letter in each brackets when you're done this item:

- [x] Code changes complete.
- [x] Code documentation complete.
- [x] Documentation for users complete (or not required, if user never sees these changes).
- [x] Manual testing done. 
- [ ] Unit-tests implemented.
- [x] Ready for merge.

zeffii commented on utils/modules/polygon_utils.py:
```diff
@@ -177,27 +178,38 @@ def pols_neighbor_num(verts, pols):
+    if pol_sides > 3:
+        f_normals = np.zeros((len(v_pols), 3), dtype=np.float64)
+        for i in range(pol_sides - 2):
+            f_normals += np.cross(v_pols[::, (1+i)%pol_sides] - v_pols[::, 0], v_pols[::, (2+i)%pol_sides] - v_pols[::, 0])
```
this is not expecting ngons / irregular / concavity ?

vicdoval replied:
Actually it  does react very similar to concave ngons to the standard bmesh calculation
![image](https://user-images.githubusercontent.com/10011941/114272579-0abcc200-9a17-11eb-841f-7a221b912b91.png)

zeffii replied:
most of the geometry produced by sverchok will go through this no problem, but for complex stuff we need a tessellator of the ngon.

if society wasn't being actively phucked as we speak i would invest more time in that myself:)

vicdoval replied:
What kind of precision can someone expect from the normal calculation of a non-planar concave polygon? :)

zeffii replied:
i'm talking about planar concave polygons. I don't see how the above code will work for those, some normals will be flipped .

zeffii replied:
it will be fine for most applications, but this is why the VDdraw has to tesselate ngons first before assigning normal derived colours.

zeffii replied:
i'd be happy to completely misunderstand this :)

vicdoval replied:
as i understand it the correct tesselation is only necessary to display the resulting triangles not to compute de normals

vicdoval replied:
![image](https://user-images.githubusercontent.com/10011941/114275000-2af17e80-9a21-11eb-8429-d9cebcd990f8.png)

zeffii replied:
ifso, then inset_specials can be done much faster too

vicdoval replied:
yes, have it almost ready. faster and with more options :D

zeffii replied:
well that's very cool! 
tested this now. I can't break it changed to seed in `ngon` :) , disregarding the invalid ngons ofcourse.

![image](https://user-images.githubusercontent.com/619340/114284645-8c314600-9a51-11eb-9474-5bcadf0e1eac.png)

[ngon test_2021_04_10_21_09.zip](https://github.com/nortikin/sverchok/files/6290767/ngon.test_2021_04_10_21_09.zip)

nice work.

vicdoval replied:
Thanks @zeffii for  understanding the joy of a  Numpy Normals algorithm.. :D

vicdoval commented:
I have just discovered I added a different algorithm Vertex Normal Algorithm 
```
"Mean Weighted Equally” (MWE) algorithm, was
introduced by Henri Gouraud[1] in 1971
```
https://www.researchgate.net/publication/220067106_A_comparison_of_algorithms_for_vertex_normal_computation
